DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -581,10 +581,16 @@ protected override void HostLaunchGame()
 
                 List<int> playerPorts = tunnelHandler.CurrentTunnel.GetPlayerPortInfo(Players.Count);
 
+                if (playerPorts.Count < Players.Count && tunnelHandler.Tunnels.Any())
+                {
+                    tunnelHandler.CurrentTunnel = tunnelHandler.Tunnels.Aggregate((i1, i2) => i1.Rating < i2.Rating ? i1 : i2);
+                    playerPorts = tunnelHandler.CurrentTunnel.GetPlayerPortInfo(Players.Count);
```

Comment:
I'm against hidden changes of tunnel server, and this stuff also makes the whole following codeblock obsolete.

Reply:
I've added logging here so that the change is no longer hidden.

Reply:
This bit of code is designed to reduce frustration when a player wants to start a game and the tunnel server is out of resources. Manual tunnel selection is an advanced option  now but it will be exclusively a power user option in the future when v3 tunnels are implemented.

The manual selection code isn't really ready to handle corner cases like this. It's a bit hacky, it should have it's own PR to handle it correctly. I don't think it's a good idea to block this PR for that reason.

Reply:
The code redundancy must be fixed either way, preferably in a way I proposed - change tunnel server silently only in case the user didn't select the tunnel manually before.

PR shouldn't introduce more inoptimal solutions code-wise, in my opinion.